projects/slap_manipulation/scripts/slap_inference.py

```diff
@@ -0,0 +1,124 @@
+import hydra
+import numpy as np
+import rospy
+import torch
+import trimesh
+from slap_manipulation.env.stretch_manipulation_env import StretchManipulationEnv
+from slap_manipulation.policy.action_prediction_module import APModule
+from slap_manipulation.policy.interaction_prediction_module import IPModule
+
+from home_robot.utils.point_cloud import show_point_cloud
+
+# TODO: move this to a constants file
+STRETCH_GRIPPER_MAX = 0.6
+
+
+def create_apm_input(raw_data, p_i):
+    raise NotImplementedError
+
+
+def create_ipm_input(
+    raw_data: dict, lang: list[str], filter_depth=False, debug=False, num_pts=8000
+):
+    input_vector = ()
+    depth = raw_data["depth"]
+    rgb = raw_data["rgb"].astype(np.float64)
+    xyz = raw_data["xyz"].astype(np.float64)
+    camera_pose = raw_data["camera_pose"]
+    proprio = raw_data["gripper_state"].astype(np.float64)
+    if proprio < 0.8 * STRETCH_GRIPPER_MAX:
+        proprio = np.array([1.0, proprio, -1.0])
+    else:
+        proprio = np.array([0.0, proprio, -1.0])
+
+    depth = depth.reshape(-1)
+    rgb = rgb.reshape(-1, 3)
+    cam_xyz = xyz.reshape(-1, 3)
+    xyz = trimesh.transform_points(cam_xyz, camera_pose)
+
+    # apply depth and z-filter for comparative distribution to training data
+    if filter_depth:
+        valid_depth = np.bitwise_and(depth > 0.1, depth < 1.0)
+        rgb = rgb[valid_depth, :]
+        xyz = xyz[valid_depth, :]
+        z_mask = xyz[:, 2] > 0.7
+        rgb = rgb[z_mask, :]
+        xyz = xyz[z_mask, :]
+
+    # get 8k points for tractable learning
+    downsample_mask = np.arange(rgb.shape[0])
+    np.random.shuffle(downsample_mask)
+    if num_pts != -1:
+        downsample_mask = downsample_mask[:num_pts]
+    rgb = rgb[downsample_mask]
+    xyz = xyz[downsample_mask]
+
+    # mean-center the point cloud
+    xyz -= xyz.mean(axis=0)
+
+    if debug:
+        show_point_cloud(xyz, rgb / 255.0, orig=np.zeros(3))
+
+    input_vector = ([rgb / 255.0], xyz, proprio, lang)
```

Comment:
we should add some sanity checks on rgb value here - make sure they arent getting quashed too low or something